ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/uniqueness/client/LedgerUniquenessCheckerClientService.kt

```diff
@@ -0,0 +1,67 @@
+package net.corda.v5.ledger.utxo.uniqueness.client
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.application.uniqueness.model.UniquenessCheckResponse
+import net.corda.v5.application.uniqueness.model.UniquenessCheckStateRef
+import java.time.Instant
+import java.util.concurrent.Future
+
+/**
+ * This service is used to request uniqueness checking. The request will be processed asynchronously
+ * and a [Future] will be returned that can be used to check the outcome. This service can be injected
+ * to either flows or other services.
+ */
+@DoNotImplement
+interface LedgerUniquenessCheckerClientService {
+
+    /**
+     * @param txId The ID of the transaction that needs processing
+     *
+     * @param inputStates A list of the input state refs that belongs to the given transaction in a
```

Comment:
We should be clear about the case sensitivity of these string inputs. (I am assuming they are case sensitive, but that in turn does mean that the StateRef.toString() has to be formally stable)

Reply:
As discussed this is using the `UniquenessCheckStateRef` interface where the toString functionality will depend on the implementation.